File: php/src/Embed.php
<?php

namespace MisarBlog;

class Embed
{
    public const EMBED_BASE = 'https://misar.blog';

    public static function url(string $username, ?string $slug = null, string $theme = 'auto'): string
    {
        $url = self::EMBED_BASE . '/' . $username;
        if ($slug !== null && $slug !== '') {
            $url .= '/' . $slug;
        }
        $url .= '/embed';
        if ($theme !== 'auto') {
            $url .= '?theme=' . urlencode($theme);
        }
        return $url;
    }
}

File: csharp/src/Embed.cs
namespace MisarBlog;

public static class Embed
{
    private const string EmbedBase = "https://misar.blog";

    public static string Url(string username, string? slug = null, string theme = "auto")
    {
        var url = string.IsNullOrEmpty(slug)
            ? $"{EmbedBase}/{username}/embed"
            : $"{EmbedBase}/{username}/{slug}/embed";

        if (!string.Equals(theme, "auto", StringComparison.OrdinalIgnoreCase))
            url += $"?theme={Uri.EscapeDataString(theme)}";

        return url;
    }
}
